Update 1.7.x nginx template

juangiordana opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.x
| Description?  | This is an effort to improve the default Nginx configuration and documentation for PrestaShop.


|| * Correct spelling errors.
|| * Shorten `location` regex for browser caching.
|| * Improve documentation to enable compression in Nginx.
|| * Remove gzip settings
|| * Enable HTTPS by default.
|| * Remove additional SSL configuration block.
|| * Remove duplicate `server_name` entry.
|| * Add `access_log` and `error_log` directives.
|| * Add `nginx` syntax highlighting to code blocks.
|| * Update list of denied directories.
|| * Improve comments in `nginx.conf`.
|| * Sort regex alphabetically.
|| * Add default `location / {...}` block.
|| * Sort AlphaImageLoader regex parameters alphabetically.
|| * Add "Access logging" section.
|| * Remove "Cloudflare / Max CDN fix" from `nginx.conf`.
|| * Remove "Force download of ..." sections from `nginx.conf`.

I plan to send more pull requests in this regard, but it would be nice for these to be merged for better readability,

juangiordana committed:
Correct spelling errors.

Correct spelling errors.

juangiordana committed:
Shorten `location` regex for browser caching.

Shorten and sort values alphabetically in `location` regex for browser caching.

juangiordana committed:
Improve documentation to enable compression in Nginx.

Improve recommendations to enable compression in Nginx.

http://nginx.org/en/docs/http/ngx_http_gzip_module.html

* Update `gzip_min_length` value to 1000. This is is the suggested value used in the [ngx_http_gzip_module](http://nginx.org/en/docs/http/ngx_http_gzip_module.html) page.

* Updated `gzip_types` to the most commonly used values, sorted alphabetically.

* Removed `gzip_http_version 1.1` since it's the default.

* Removed `gzip_buffers` since sane (and same) values are provided by default.

* Removed `gzip_static` directive since it's not commonly used and pollutes the provided example.

juangiordana committed:
Remove gzip settings

Remove gzip settings from the main Nginx configuration example as its
usage may vary on different HTTP/HTTPS scenarios and provide a link to
the internal Nginx optimization article that contains improved gzip
settings.

juangiordana committed:
Enable HTTPS by default.

Change default `listen` directives so HTTPS is enabled by default as
it's the most common configuration nowadays, specially for e-commerce
websites.

juangiordana committed:
Remove additional SSL configuration block.

Remove additional SSL configurations that pollutes the example and may
vary across different server configurations.

juangiordana committed:
Remove duplicate `server_name` entry.

Remove duplicate `server_name` entry.

juangiordana committed:
Add `access_log` and `error_log` directives.

Add `access_log` and `error_log` directives.

juangiordana committed:
Add `nginx` syntax highlighting to code blocks.

Add markdown `nginx` syntax highlighting to Nginx code blocks.

juangiordana committed:
Update list of denied directories.

Update list of denied directories.

- Add `var`.
- Remove `tests-legacy`.

juangiordana committed:
Improve comments in `nginx.conf`.

Improve comments in the `nginx.conf` example.

juangiordana committed:
Sort regex alphabetically.

Sort regex parameters alphabetically.

juangiordana committed:
Add default `location / {...}` block.

Add default `location / {...}` block and move `try_files` directive
from a `server` context to this default `location` context.

> The try_files directive is mostly like every other action directive
> mentioned above, the difference is that if placed in server context
> nginx actually creates a pseudo-location that is the least specific
> location possible. That means if a request matches a defined location
> the try_files directive will not be executed. This means that if you
> have location / defined then you have a location that matches every
> possible request and as such try_files will never actually execution.
> Therefore always place try_files in location context instead of
> server context if at all possible.

[Understanding the Nginx Configuration Inheritance Model](https://blog.martinfjordvald.com/understanding-the-nginx-configuration-inheritance-model/).

Use `$is_args$args` in the `try_files` directive as suggested in the
[Symfony documentation](https://symfony.com/doc/current/setup/web_server_configuration.html#web-server-nginx).

juangiordana committed:
Sort AlphaImageLoader regex parameters alphabetically.

Sort AlphaImageLoader regex parameters alphabetically.

PierreRambaud commented on basics/installation/nginx.md:
```diff
@@ -134,54 +92,53 @@ server {
-    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|tests-legacy|tools|translations|travis-scripts|vendor|var)/ {
+
+    # Source code directories.
+    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|tools|translations|travis-scripts|var|vendor)/ {
```
```suggestion
    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|translations|var|vendor)/ {
```

There are no tools or travis-scripts in recent PrestaShop installation (or in releases)

PierreRambaud commented on basics/installation/nginx.md:
```diff
@@ -6,123 +6,81 @@ weight: 25
-    # Cloudflare / Max CDN fix
-    location ~* \.(eot|otf|ttf|woff(?:2)?)$ {
+    # Cloudflare / Max CDN fix.
+    location ~* \.(eot|otf|ttf|woff|woff2)$ {
```
```suggestion
    location ~* \.(eot|otf|ttf|woff2?)$ {
```

Should be enough

PierreRambaud commented on scale/webservers/nginx.md:
````diff
@@ -4,35 +4,62 @@ title: Optimize your nginx configuration
-```
-location ~* \.(?:jpg|jpeg|gif|png|ico|css|woff2)$ {
+```nginx
+location ~* \.(?:css|gif|ico|jpe?g|png|woff2)$ {
````
```suggestion
location ~* \.(?:css|eot|gif|ico|jpe?g|otf|png|ttf|woff2?)$ {
```
It's maybe better, wdyt?

PierreRambaud reviewed (commented):
Great contribution, just few comments ^^

juangiordana committed:
Update list of denied directories.

Update list of denied directories.

- Remove `tools`.
- Remove `travis-scripts`.

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

juangiordana committed:
Simplify `location` regex.

Simplify "Cloudflare / Max CDN fix." `location` regex.

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

juangiordana committed:
Improve `location` regex for browser caching.

Improve `location` regex for browser caching.

Improve `location` regex for browser caching by adding the following file extensions: `eot`, `otf`, `ttf`, `woff`, `woff2`.

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

juangiordana commented:
Thanks! More are on the way.

juangiordana committed:
Add "Access logging" section.

Add "Access logging" section to the "Optimize your Nginx configuration"
article.

Remove these directives from the `nginx.conf` configuration example as
this may vary on different PrestaShop scenarios and pollute the
simplicity of the provided example.

juangiordana committed:
Remove "Cloudflare / Max CDN fix" from `nginx.conf`

Remove "Cloudflare / Max CDN fix" from `nginx.conf` as its use is not
required in a basic PrestaShop installation.

juangiordana committed:
Remove "Force download of ..." sections from `nginx.conf`

Remove "Force download of ..." sections from the main `nginx.conf`
example as their usage may vary across different PrestaShop
installations and hinder the simplicity of the provided example, i.e.,
the file already provides a `location /upload` section and these
directives could be specified in that section.

PierreRambaud commented on basics/installation/nginx.md:
```diff
@@ -134,54 +60,53 @@ server {
-    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|tests-legacy|tools|translations|travis-scripts|vendor|var)/ {
+
+    # Source code directories.
+    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|translations|var|vendor)/ {
```
```suggestion
    location ~ ^/(app|bin|cache|classes|config|controllers|docs|localization|override|src|tests|tools|translations|var|vendor)/ {
```

My bad tools is in the released zip :facepalm:

PierreRambaud committed:
Deny tools directory

matthieu-rolland commented:
thank you @juangiordana !

juangiordana commented:
Thanks both! There are more are on their way.

matks commented:
Hi @juangiordana and @PierreRambaud thanks for the contribution ! Could you consider backporting this to branch 8.x ?

juangiordana commented:
Hi Mathiew,

You're welcome. I'm still working on some important improvements. Once I
have them all committed I'll do that for sure.

Best

On Thu, Oct 7, 2021 at 5:55 PM Mathieu Ferment ***@***.***>
wrote:

> Hi @juangiordana <https://github.com/juangiordana> and @PierreRambaud
> <https://github.com/PierreRambaud> thanks for the contribution ! Could
> you consider backporting this to branch 8.x ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PrestaShop/docs/pull/1166#issuecomment-938148725>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAKRQXCWLGYA4CBRAXG4ZDUFYCKZANCNFSM5FJ6K4EA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Juan Fco. Giordana
+54 9 342 521 1004

matks commented:
Great to hear ;) thanks
